File: bench/polyglot/go/string_ops.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func mainWorkload() int64 {
	var checksum int64
	for i := 0; i < 4000; i++ {
		token := "user-" + strconv.Itoa(i) + "-record"
		upper := strings.ToUpper(token)
		lower := strings.ToLower(upper)
		line := lower + "|" + token + "|segment-" + strconv.Itoa(i%97)
		parts := strings.Split(line, "|")
		rebuilt := parts[0] + ";" + parts[1] + ";" + parts[2]
		replaced := strings.ReplaceAll(rebuilt, "user-", "member-")
		padded := "  " + replaced + "  "
		trimmed := strings.TrimSpace(padded)
		checksum += int64(len(trimmed)) + int64(len(parts))
		if strings.Contains(trimmed, "member-") {
			checksum += 3
		}
		if strings.HasPrefix(trimmed, "member-") {
			checksum += 7
		}
		if strings.HasSuffix(trimmed, "7") {
			checksum += 11
		}
		if i%5 == 0 {
			checksum += int64(len(strings.ReplaceAll(line, "e", "E")))
		}
	}
	return checksum
}

func main() {
	fmt.Println(mainWorkload())
}

File: bench/polyglot/ruby/string_ops.rb
def main_workload
  checksum = 0
  i = 0
  while i < 4000
    token = "user-" + i.to_s + "-record"
    upper = token.upcase
    lower = upper.downcase
    line = lower + "|" + token + "|segment-" + (i % 97).to_s
    parts = line.split("|")
    rebuilt = parts[0] + ";" + parts[1] + ";" + parts[2]
    replaced = rebuilt.gsub("user-", "member-")
    padded = "  " + replaced + "  "
    trimmed = padded.strip
    checksum += trimmed.length + parts.length
    checksum += 3 if trimmed.include?("member-")
    checksum += 7 if trimmed.start_with?("member-")
    checksum += 11 if trimmed.end_with?("7")
    checksum += line.gsub("e", "E").length if i % 5 == 0
    i += 1
  end
  checksum
end

puts main_workload
